src/pages/index.js

```diff
@@ -0,0 +1,309 @@
+}
+
+// Open Modal
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+}
+// Close Modal
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+
+  // Adding enableButton here
+  const submitButton = modal.querySelector(settings.submitButtonSelector)
+  if (submitButton) enableButton(submitButton, settings.inactiveButtonClass)
+}
+
+// Escape Key Close
+function handleEscapeKey(evt) {
+  if (evt.key === "Escape") {
+    const currentModal = document.querySelector(".modal_opened");
+    if (currentModal) {
+      closeModal(currentModal);
+    }
+  }
+}
+
+// Iterate over modal overlay to add click outside of content
+overlays.forEach((modal) => {
+  modal.addEventListener("click", (evt) => {
+    if (evt.target === modal && modal.classList.contains("modal_opened")) {
+      closeModal(modal);
+    }
+  });
+});
+
+// Edit Profile
+// ---------------------------
+function handleEditFormSubmit(evt) {
+  handleSubmit(updateEditProfileForm, evt, 'Saving. . .', settings.inactiveButtonClass);
+}
+
+function updateEditProfileForm() {
+
+  return api.editUserInfo({
+    name: editModalNameInput.value,
+    about: editModalDescriptionInput.value
+  })
+    .then((userInfo) => {
+
+      profileName.textContent = userInfo.name;
+      profileDescription.textContent = userInfo.about;
+      closeModal(editProfileModal);
+      // disableButton(editFormElement.querySelector(settings.submitButtonSelector), settings.inactiveButtonClass)
+
+    })
+    .catch(console.error)
+
+}
+// -------------------------------
+
+// New Post --------------------
+function handleAddCardSubmit(evt) {
+  handleSubmit(addCardReq, evt, "Saving. . .")
+}
+
+function addCardReq() {
+  return api.addNewCard({name: cardNameInput.value, link: cardLinkInput.value})
+    .then(renderCard)
+    .then(() => closeModal(cardModal));
+}
+// ------------------------------
+
+// Avatar  - - - - - - - -
+function handleAvatarSubmit(evt) {
+  handleSubmit(addAvatarReq, evt, "Saving. . .",settings.inactiveButtonClass)
+}
+
+function addAvatarReq(){
+  return api.editUserAvatar({ avatar: avatarLinkInput.value })
+    .then((userPic) => {
+      profileImage.src = userPic.avatar;
+      closeModal(avatarModal);
+      // disableButton(editFormElement.querySelector(settings.submitButtonSelector), settings.inactiveButtonClass)
+    })
+
+}
+// - - - - - - - -- - - - - - - -
+
+// Delete - - - - - - - - -
+
+function handleDeleteCard(cardElement, cardId) {
+  selectedCard = cardElement;
+  selectedCardId = cardId;
+  openModal(deleteModal);
+}
+// function handleDeleteSubmit(evt) {
+//   handleSubmit(addDeleteReq, evt, "Deleting. . . ");
+// }
+
+function handleDeleteSubmit(evt) {
+  evt.preventDefault();
+
+  const submitButton = evt.submitter;
+  if (!submitButton) return;
+
+  renderLoading(true, submitButton, "Deleting...");
```

Comment:
`renderLoading` accepts 4 params, and the 4th one is `loadingText`

`"Deleting...` should be the 4th here

So, it should be like this:


```js
  renderLoading(true, submitButton, "Delete", "Deleting...");


```